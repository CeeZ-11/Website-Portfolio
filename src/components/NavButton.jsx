export default function NavButton({ navButtons, activeTab, setActiveTab }) {
  return navButtons.map((button) => (
    <li key={button.id}>
      <button
        onClick={() => setActiveTab(`${button.id}`)}
        className={activeTab === `${button.id}` ? "active" : ""}
      >
        {button.name}
      </button>
    </li>
  ));
}
