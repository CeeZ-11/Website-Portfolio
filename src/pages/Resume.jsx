import PageTitle from "../components/PageTitle";
import { useState } from "react";

export default function Resume() {
  const [activeTab, setActiveTab] = useState("languages");

  return (
    <>
      <PageTitle name="Resume" />
      <div className="experience">
        <div className="exp-title">
          <h2>Experience</h2>
          <div className="exp-info-container">
            <div className="exp-line"></div>
            <div className="exp-details">
              <ul>
                <li>
                  <h4>Web Consultant (Freelance) </h4>
                  <h5>OCT 2024 - PRESENT</h5>
                  <ul>
                    <li>
                      Maintained and managed client websites hosted on GoDaddy
                      and built on WordPress, ensuring reliability and uptime.
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div className="education">
        <div className="edu-title">
          <h2>Education</h2>
          <div className="edu-info-container">
            <div className="edu-line"></div>
            <div className="edu-details">
              <ul>
                <li>
                  <h4>The Odin Project Curriculum</h4>
                  <h5>Nov 2023 - Present</h5>
                  <ul>
                    <li>
                      Comprehensive coursework covering full-stack web
                      development.
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div className="additional-info">
        <h2>Additional Information</h2>
        <nav className="info-nav">
          <ul>
            <li>
              <button
                onClick={() => setActiveTab("languages")}
                className={activeTab === "languages" ? "active" : ""}
              >
                Languages
              </button>
            </li>
            <li>
              <button
                onClick={() => setActiveTab("certifications")}
                className={activeTab === "certifications" ? "active" : ""}
              >
                Certifications
              </button>
            </li>
            <li>
              <button
                onClick={() => setActiveTab("trainings")}
                className={activeTab === "trainings" ? "active" : ""}
              >
                Trainings & Seminars
              </button>
            </li>
          </ul>
        </nav>
        <div className="info-content">
          {activeTab === "languages" && (
            <>
              <h4> Languages </h4>
              <ul>
                <li> English - B2(Upper Intermediate)</li>
                <li>Tagalog - Native or Bilingual</li>
              </ul>
            </>
          )}
          {activeTab === "certifications" && (
            <>
              <h4> Certifications </h4>
              <ul>
                <li>SAP Business One Certification (Jan 2016)</li>
                <li>Apple Teacher Certification (Oct 2019)</li>
              </ul>
            </>
          )}
          {activeTab === "trainings" && (
            <>
              <h4> Trainings & Seminars </h4>
              <ul>
                <li>JobStreet.com â€“ Jumpstart Session (2016)</li>
                <li>Google I/O Seminar (2016)</li>
              </ul>
            </>
          )}
        </div>
      </div>
    </>
  );
}
