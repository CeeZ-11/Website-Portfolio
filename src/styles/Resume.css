.additional-info {
  display: flex;
  gap: 15px;
  flex-direction: column;
  justify-content: center;
}

.exp-details {
  position: relative;
}

.exp-info-container .exp-details ul li .resume-ul-desc .resume-desc {
  padding: 0;
  list-style-type: none;
}

.exp-info-container .exp-details ul li .resume-ul-desc {
  padding: 0;
}

.exp-info-container .exp-details > ul {
  list-style: none;
  padding-left: 0;
  display: flex;
  flex-direction: column;
  gap: 50px;
}

.exp-info-container .exp-details > ul > li {
  display: flex;
  padding-left: 45px;
  position: relative;
  flex-direction: column;
  gap: 10px;
}

.exp-info-container .exp-details > ul > li::before {
  content: "●";
  color: rgb(255, 174, 107);
  position: absolute;
  left: 2px;
  text-shadow: 0 0 8px rgba(255, 115, 0, 0.8), 0 0 12px rgba(255, 115, 0, 0.6),
    0 0 16px rgba(255, 115, 0, 0.4);
}

.exp-info-container .exp-details > ul::before {
  content: "";
  position: absolute;
  left: 7px;
  top: 0;
  width: 1px;
  height: 100%;
  background-color: rgb(255, 255, 255, 0.5);
}

.additional-info nav ul li button {
  background: none;
  color: inherit;
  padding: 10px;
  border: none;
  cursor: pointer;
}

.additional-info nav ul li button.active {
  color: #ff7300;
  text-shadow: 0 0 6px rgba(255, 115, 0, 0.5), 0 0 10px rgba(255, 115, 0, 0.4);
}

.info-container {
  background: rgba(255, 255, 255, 0.02);
  padding: 1rem 2rem 2rem 2rem;
  border-radius: 15px;
  box-shadow: 0 0 3px rgba(255, 255, 255, 0.8);
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.info-content {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.info-title {
  text-align: start;
  width: 100%;
}

.resume-icons {
  width: 15px;
  height: 15px;
}

.resume-heads {
  display: flex;
  gap: 30px;
  justify-content: start;
  align-items: center;
}

.exp-title,
.edu-title {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

@media (max-width: 1300px) {
  .exp-info-container .exp-details > ul > li::before {
    left: 3px;
  }
}

@media (max-width: 1000px) {
  .exp-info-container .exp-details > ul > li::before {
    left: 4px;
  }
}

@media (max-width: 500px) {
  .info-nav ul li button {
    padding: 10px;
    font-size: 10px;
  }
}

@media (max-width: 340px) {
  .info-nav ul li button {
    padding: 5px;
    font-size: 8px;
  }
}

@media (max-width: 300px) {
  .info-nav ul li button {
    padding: 3px;
    font-size: 6px;
  }
}
