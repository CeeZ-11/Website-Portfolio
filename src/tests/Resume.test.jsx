import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import Resume from "../pages/Resume";
import NavButton from "../components/NavButton";

describe("Resume Page", () => {
  it("renders the Contact page correctly", () => {
    const { asFragment } = render(<Resume />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 'Experience' section", () => {
    render(<Resume />);
    expect(screen.getByText("Experience")).toBeInTheDocument();
    expect(screen.getByText("Web Consultant (Freelance)")).toBeInTheDocument();
    expect(screen.getByText("OCT 2024 - PRESENT")).toBeInTheDocument();
    expect(
      screen.getByText(
        "Managed and maintained WordPress websites on GoDaddy, handling updates, design adjustments, performance optimization, and client collaboration."
      )
    ).toBeInTheDocument();
  });

  it("renders 'Education' section", () => {
    render(<Resume />);
    expect(screen.getByText("Education")).toBeInTheDocument();
    expect(screen.getByText("The Odin Project")).toBeInTheDocument();
    expect(screen.getByText("Nov 2023 - Present")).toBeInTheDocument();
    expect(
      screen.getByText(
        "Completed full-stack web development coursework, gaining hands-on experience with modern technologies and best practices."
      )
    ).toBeInTheDocument();
  });

  it("renders 'Certifications' after button click", async () => {
    const navButtons = [
      { id: "languages", name: "Languages" },
      { id: "certifications", name: "Certifications" },
      { id: "trainings", name: "Trainings & Seminars" },
    ];

    const setActiveTab = vi.fn();
    const activeTab = "certifications";

    const user = userEvent.setup();

    render(
      <NavButton
        navButtons={navButtons}
        activeTab={activeTab}
        setActiveTab={setActiveTab}
      />
    );

    const button = screen.getByRole("button", { name: "Certifications" });

    await user.click(button);

    expect(screen.getByText("Certifications")).toBeInTheDocument();
  });
});
