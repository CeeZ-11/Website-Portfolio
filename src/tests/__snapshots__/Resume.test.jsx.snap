// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resume Page > renders the Contact page correctly 1`] = `
<DocumentFragment>
  <div
    class="about-title"
  >
    <h1>
      Resume
    </h1>
    <hr
      class="header-line"
    />
  </div>
  <div
    class="experience"
  >
    <div
      class="exp-title"
    >
      <div
        class="resume-heads"
      >
        <svg
          class="resume-icons"
          enable-background="new 0 0 64 64"
          fill="#e3e3e3"
          id="Layer_1"
          stroke="#e3e3e3"
          version="1.0"
          viewBox="0 0 64 64"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="SVGRepo_bgCarrier"
            stroke-width="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
             
            <path
              d="M60,13H48V4c0-2.211-1.789-4-4-4H20c-2.211,0-4,1.789-4,4v9H4c-2.211,0-4,1.789-4,4v43c0,2.211,1.789,4,4,4 h56c2.211,0,4-1.789,4-4V17C64,14.789,62.211,13,60,13z M18,4c0-1.104,0.896-2,2-2h24c1.104,0,2,0.896,2,2v9h-2V5 c0-0.553-0.447-1-1-1H21c-0.553,0-1,0.447-1,1v8h-2V4z M42,6v7H22V6H42z M62,60c0,1.104-0.896,2-2,2H4c-1.104,0-2-0.896-2-2V42h10v5 c0,0.553,0.447,1,1,1h6c0.553,0,1-0.447,1-1v-5h24v5c0,0.553,0.447,1,1,1h6c0.553,0,1-0.447,1-1v-5h10V60z M14,46V36h4v10H14z M46,46V36h4v10H46z M62,40H52v-5c0-0.553-0.447-1-1-1h-6c-0.553,0-1,0.447-1,1v5H20v-5c0-0.553-0.447-1-1-1h-6 c-0.553,0-1,0.447-1,1v5H2V17c0-1.104,0.896-2,2-2h56c1.104,0,2,0.896,2,2V40z"
              fill="#e3e3e3"
            />
             
          </g>
        </svg>
        <h2>
          Experience
        </h2>
      </div>
      <div
        class="exp-info-container"
      >
        <div
          class="exp-line"
        />
        <div
          class="exp-details"
        >
          <ul>
            <li>
              <div
                class="exp-details-head"
              >
                <h4>
                  Web Consultant (Freelance)
                </h4>
                <h5>
                  OCT 2024 - PRESENT
                </h5>
              </div>
              <ul
                class="resume-ul-desc"
              >
                <li
                  class="resume-desc"
                >
                  Managed and maintained WordPress websites on GoDaddy, handling updates, design adjustments, performance optimization, and client collaboration.
                </li>
              </ul>
            </li>
            <li>
              <div
                class="exp-details-head"
              >
                <h4>
                  Monitor Engineer, MID Construction Group
                </h4>
                <h5>
                  MAY 2024 - OCT 2024
                </h5>
              </div>
              <ul
                class="resume-ul-desc"
              >
                <li
                  class="resume-desc"
                >
                  Monitored 45+ Google Meet sessions, tracked attendance, enforced rules, provided technical support, evaluated performance, and optimized workflows.
                </li>
              </ul>
            </li>
            <li>
              <div
                class="exp-details-head"
              >
                <h4>
                  Logistics Coordinator, Rulewave
                </h4>
                <h5>
                  September 2021 - June 2023
                </h5>
              </div>
              <ul
                class="resume-ul-desc"
              >
                <li
                  class="resume-desc"
                >
                  Optimized supply chain operations, managed customer inquiries, analyzed reports, tracked shipments, and improved logistics processes for efficiency.
                </li>
              </ul>
            </li>
            <li>
              <div
                class="exp-details-head"
              >
                <h4>
                  Professor, University of St. La Salle
                </h4>
                <h5>
                  July 2017 - April 2021
                </h5>
              </div>
              <ul
                class="resume-ul-desc"
              >
                <li
                  class="resume-desc"
                >
                  Taught HTML, CSS, PHP, and JavaScript, while also facilitating research methodology sessions to enhance students' programming and research skills.
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    class="education"
  >
    <div
      class="edu-title"
    >
      <div
        class="resume-heads"
      >
        <svg
          class="resume-icons"
          enable-background="new 0 0 64 64"
          fill="#e3e3e3"
          id="Layer_1"
          version="1.0"
          viewBox="0 0 64 64"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="SVGRepo_bgCarrier"
            stroke-width="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
             
            <g>
               
              <path
                d="M24.008,32.038L9.541,27.904c-0.527-0.146-1.084,0.155-1.236,0.688c-0.151,0.53,0.156,1.084,0.688,1.236 l14.467,4.134C23.551,33.987,23.643,34,23.734,34c0.435,0,0.835-0.286,0.961-0.726C24.847,32.744,24.539,32.19,24.008,32.038z"
                fill="#e3e3e3"
              />
               
              <path
                d="M24.008,39.038L9.541,34.905c-0.527-0.146-1.084,0.155-1.236,0.688c-0.151,0.531,0.156,1.084,0.688,1.236 l14.467,4.133C23.551,40.987,23.643,41,23.734,41c0.435,0,0.835-0.286,0.961-0.726C24.847,39.743,24.539,39.19,24.008,39.038z"
                fill="#e3e3e3"
              />
               
              <path
                d="M24.008,25.038L9.541,20.904c-0.527-0.146-1.084,0.155-1.236,0.688c-0.151,0.53,0.156,1.084,0.688,1.236 l14.467,4.134C23.551,26.987,23.643,27,23.734,27c0.435,0,0.835-0.286,0.961-0.726C24.847,25.744,24.539,25.19,24.008,25.038z"
                fill="#e3e3e3"
              />
               
              <path
                d="M24.008,18.038L9.541,13.904c-0.527-0.146-1.084,0.155-1.236,0.688c-0.151,0.53,0.156,1.084,0.688,1.236 l14.467,4.134C23.551,19.987,23.643,20,23.734,20c0.435,0,0.835-0.286,0.961-0.726C24.847,18.744,24.539,18.19,24.008,18.038z"
                fill="#e3e3e3"
              />
               
              <path
                d="M39.963,33.962c0.092,0,0.184-0.013,0.275-0.038l14.467-4.134c0.531-0.152,0.839-0.706,0.688-1.236 c-0.152-0.532-0.708-0.832-1.236-0.688L39.689,32c-0.531,0.152-0.839,0.706-0.688,1.236C39.128,33.676,39.528,33.962,39.963,33.962 z"
                fill="#e3e3e3"
              />
               
              <path
                d="M54.459,34.905l-14.467,4.133c-0.531,0.152-0.839,0.705-0.688,1.236C39.431,40.714,39.831,41,40.266,41 c0.092,0,0.184-0.013,0.275-0.038l14.467-4.133c0.531-0.152,0.839-0.705,0.688-1.236C55.543,35.061,54.987,34.761,54.459,34.905z"
                fill="#e3e3e3"
              />
               
              <path
                d="M54.459,20.904l-14.467,4.134c-0.531,0.152-0.839,0.706-0.688,1.236C39.431,26.714,39.831,27,40.266,27 c0.092,0,0.184-0.013,0.275-0.038l14.467-4.134c0.531-0.152,0.839-0.706,0.688-1.236C55.543,21.06,54.987,20.758,54.459,20.904z"
                fill="#e3e3e3"
              />
               
              <path
                d="M54.459,13.904l-14.467,4.134c-0.531,0.152-0.839,0.706-0.688,1.236C39.431,19.714,39.831,20,40.266,20 c0.092,0,0.184-0.013,0.275-0.038l14.467-4.134c0.531-0.152,0.839-0.706,0.688-1.236C55.543,14.06,54.987,13.76,54.459,13.904z"
                fill="#e3e3e3"
              />
               
              <path
                d="M63.219,0.414c-0.354-0.271-0.784-0.413-1.221-0.413c-0.172,0-0.345,0.022-0.514,0.066L32,7.93L2.516,0.067 c-0.17-0.045-0.343-0.066-0.515-0.066c-0.437,0-0.866,0.142-1.22,0.413C0.289,0.793,0,1.379,0,2v49.999 c0,0.906,0.609,1.699,1.484,1.933l25.873,6.899C28.089,62.685,29.887,64,32,64s3.911-1.315,4.643-3.169l25.873-6.899 C63.391,53.698,64,52.905,64,51.999V2C64,1.379,63.711,0.793,63.219,0.414z M32,54c0.173,0,0.347-0.022,0.516-0.067L62,46.07v1.954 l-30,7.941L2,48.024V46.07l29.484,7.862C31.653,53.978,31.827,54,32,54z M1.998,2.001c0,0,0.001,0,0.003,0V2L31,9.733v42L2,44 L1.998,2.001z M34.979,59.205c-0.079,1.143-0.785,2.111-1.788,2.546l-0.676,0.181c-0.169,0.045-0.343,0.067-0.516,0.067 s-0.347-0.022-0.516-0.067l-0.676-0.181c-1.003-0.435-1.709-1.403-1.788-2.546L2,51.999v-1.906l29.744,7.874 C31.828,57.989,31.914,58,32,58s0.172-0.011,0.256-0.033L62,50.093v1.906L34.979,59.205z M33,51.733v-42L62,2v42L33,51.733z"
                fill="#e3e3e3"
              />
               
            </g>
             
          </g>
        </svg>
        <h2>
          Education
        </h2>
      </div>
      <div
        class="edu-info-container"
      >
        <div
          class="edu-line"
        />
        <div
          class="edu-details"
        >
          <ul>
            <li>
              <h4>
                The Odin Project
              </h4>
              <h5>
                Nov 2023 - Present
              </h5>
              <ul
                class="edu-ul-desc"
              >
                <li
                  class="edu-desc"
                >
                  Completed full-stack web development coursework, gaining hands-on experience with modern technologies and best practices.
                </li>
              </ul>
            </li>
            <li>
              <h4>
                University of St. La Salle 
              </h4>
              <h5>
                Jun 2013 - Mar 2017
              </h5>
              <ul
                class="edu-ul-desc"
              >
                <li
                  class="edu-desc"
                >
                  Major in Web and Mobile Development, with a thesis on Information Management System for Sacred Heart Shrine Lupit Church.
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    class="additional-info"
  >
    <div
      class="info-title"
    >
      <h2>
        Additional Information
      </h2>
    </div>
    <div
      class="info-container"
    >
      <nav
        class="info-nav"
      >
        <ul>
          <li>
            <button
              class="active"
            >
              Languages
            </button>
          </li>
          <li>
            <button
              class=""
            >
              Certifications
            </button>
          </li>
          <li>
            <button
              class=""
            >
              Trainings & Seminars
            </button>
          </li>
        </ul>
      </nav>
      <div
        class="info-content"
      >
        <h4>
           Languages 
        </h4>
        <ul>
          <li>
             English - B2(Upper Intermediate)
          </li>
          <li>
            Tagalog - Native or Bilingual
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
